### ISSUE TITLE
ruby 2.4.1 install fails without error

### ISSUE THREAD

**Issue by perlun** [rvm/rvm#4007]
This is very similar to #3686, it feels like it has come back.

The problem reported there has _reappeared_ with rvm head:

```shell
$ rvm install 2.4.1
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.12/x86_64/ruby-2.4.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
```

It doesn't compile or install it, and it gives no errors.

When I added the `-j 8` it started downloading and compiling it... Is there something weird going on with the CPU detection again?

**Comment by mpapis** [comment]
can you show us gist of `rvm --trace install 2.4.1 2>&1 | tee big.log`

**Comment by perlun** [comment]
@mpapis Hmm, I installed it with `-j 8` and now that I uninstalled it it seems to work... so I must have unbroken it. 😊 

I suggest we keep this open for a while, if someone else has a repro we can dig in more but otherwise, we can just close it as "can't reproduce".

**Comment by pkuczynski** [comment]
No point to keep it open as it gets on my way when planning work. We can always re-open when we get more detailed reports.

**Comment by perlun** [comment]
Fine with me, thanks anyway.

**Comment by datSunday** [comment]
I'm having the same issue as @perlun 
```
➜  workspace rvm install 2.4.1
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.12/x86_64/ruby-2.4.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
__rvm_detect_max_threads:cd:21: string not in pwd: /Users/sunday/workspace/contributions
```

and before I go ahead to run with `-j 8`, I ran `rvm --trace install 2.4.1 2>&1 | tee big.log` as suggested by @mpapis and [here is the gist](https://gist.github.com/datSunday/2c32e947a992066d21002e38ebb1a6cd), in case it's useful.

I'll go ahead and use `-j 8` now and report back with updates.

**Comment by datSunday** [comment]
Using the `-j` flag worked for me as well... here is the trace:

```
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.12/x86_64/ruby-2.4.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
Found user configured '-j' flag in 'rvm_make_flags', please note that RVM can detect number of CPU threads and set the '-j' flag automatically if you do not set it.
Installing Ruby from source to: /Users/sunday/.rvm/rubies/ruby-2.4.1, this may take a while depending on your cpu(s)...
ruby-2.4.1 - #downloading ruby-2.4.1, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 11.9M  100 11.9M    0     0   103k      0  0:01:58  0:01:58 --:--:--  121k
ruby-2.4.1 - #extracting ruby-2.4.1 to /Users/sunday/.rvm/src/ruby-2.4.1 - please wait
ruby-2.4.1 - #applying patch /Users/sunday/.rvm/patches/ruby/2.4.1/random_c_using_NR_prefix.patch - please wait
ruby-2.4.1 - #configuring - please wait
ruby-2.4.1 - #post-configuration - please wait
ruby-2.4.1 - #compiling - please wait
ruby-2.4.1 - #installing - please wait
ruby-2.4.1 - #making binaries executable - please wait
ruby-2.4.1 - #downloading rubygems-2.6.13
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  750k  100  750k    0     0  18245      0  0:00:42  0:00:42 --:--:-- 97781
ruby-2.4.1 - #extracting rubygems-2.6.13 - please wait
ruby-2.4.1 - #removing old rubygems - please wait
$LANG was empty, setting up LANG=en_US, if it fails again try setting LANG to something sane and try again.
ruby-2.4.1 - #installing rubygems-2.6.13 - please wait
ruby-2.4.1 - #gemset created /Users/sunday/.rvm/gems/ruby-2.4.1@global
ruby-2.4.1 - #importing gemset /Users/sunday/.rvm/gemsets/global.gems - please wait
ruby-2.4.1 - #generating global wrappers - please wait
ruby-2.4.1 - #gemset created /Users/sunday/.rvm/gems/ruby-2.4.1
ruby-2.4.1 - #importing gemsetfile /Users/sunday/.rvm/gemsets/default.gems evaluated to empty gem list
ruby-2.4.1 - #generating default wrappers - please wait
ruby-2.4.1 - #adjusting #shebangs for (gem irb erb ri rdoc testrb rake).
Install of ruby-2.4.1 - #complete 
ruby_install_warnings:cd:4: string not in pwd: /Users/sunday/workspace/contributions
Ruby was built without documentation, to build it run: rvm docs generate-ri
```

I also checked my `$HOME/.rvmrc` if I had `rvm_make_flags` set to something default, but it wasn't there.
